lsp-methods.el

```diff
@@ -1289,6 +1289,61 @@ export interface MarkupContent {
   (interactive)
   (lsp--text-document-hover-string))
 
+(defvar-local lsp--current-signature-help-request-id nil)
+
+(defun lsp--text-document-signature-help ()
+  "interface SignatureHelp {
+signatures: SignatureInformation[];
+activeSignature?: number;
+activeParameter?: number;
+};
+
+interface SignatureInformation {
+label: string;
+documentation?: string | MarkupContent;
+parameters?: ParameterInformation[];
+};
+
+interface ParameterInformation {
+label: string;
+documentation?: string | MarkupContent;
+};
+
+interface MarkupContent {
+kind: MarkupKind;
+value: string;
+};
+
+type MarkupKind = 'plaintext' | 'markdown';"
+  (lsp--cur-workspace-check)
+  (when lsp--current-signature-help-request-id
+    (lsp--cancel-request lsp--current-signature-help-request-id))
+  (let (bounds body)
+    (when (symbol-at-point)
```

Comment:
I am testing the method with JDT and it works only if it is invoked within a function invocation. @vibhavp do we need to fix this function or lsp-java should implement it's own version?

Reply:
Isn't that the intended functionality for signatureHelp?

Reply:
@vibhavp
Assuming that you have the following code and the cursor is at |

> foo.bar(param1, |);

In this case, the LSP should display signature help for the method bar but since (symbol-at-point) will return false the signature help method will not run at all.